### PR TITLE
DataArray and DataView bindings

### LINKED ISSUE #11: Complete bindings for DataArray
Implement all data related methods on `DataArray` such as `getData()` and `setData()`. For this task a good representation for ND arrays.

### PR DESCRIPTION
**Pull request by sujithvm**
Issue fix #11 . Thank you

### CODE REVIEW

**Comment by sujithvm** [comment]
@stoewer Can you please review this pull request? Thank you

File: src/main/java/org/gnode/nix/DataArray.java [hunk 1]
```diff
@@ -646,6 +649,218 @@ public void setPolynomCoefficients(double[] polynomCoefficients) {
...
+     * @param count  dimensions
+     * @param offset offset
+     * @return double array
+     */
+    public double[] getDataAsDoubleArray(NDSize count, NDSize offset) {
+        double[] data = new double[(int) count.getElementsProduct()];
+        getDataDirect(DataType.Double, data, count, offset);
+        return data;
+    }
+
```
Comment (stoewer) [inline src/main/java/org/gnode/nix/DataArray.java]: I would prefer if the getters would resemble more the setters and have the array as an input parameter:

```java
public void getData(int[] data, NDSize count, NDSize offset)

public void getData(int[] data, NDSize count, NDSize offset, 
                    int dataType)

// same for other primitive types
```

Comment (sujithvm) [inline src/main/java/org/gnode/nix/DataArray.java]: @stoewer Should dataType parameter also be there?

**Comment by sujithvm** [comment]
@stoewer I have updated the getData functions. Please check when you are free. Thank you

**Comment by stoewer** [comment]
Looks good to me :+1:

**Pull request merged**